backend/routes/goals.py

```
@@ -278,12 +271,50 @@ def should_show_personal_goal_button():
+        )
         return (
             jsonify({"error": "An error occurred while checking allow personal goals"}),
             500,
-        )
\ No newline at end of file
+        )
+
+
+@goals_bp.route("/reward-weekly-completion-goal", methods=["PUT"])
+@jwt_required()
+def add_weekly_completion_goal_gems():
```

Ruff (0.8.2)

Missing return type annotation for public function `add_weekly_completion_goal_gems`

(ANN201)